crt/memory.c

```diff
@@ -24,8 +24,11 @@ void *
 __wrap_realloc(void *ptr, size_t size) {
    void *realloc_ptr = __wrap_malloc(size);
```

Comment:
Why not just put this at the top, I think it makes the intent a lot more clear, and avoids doing an alloc/free for no reason.
```c++
if (!ptr) {
   return nullptr;
}
```